query.go

```diff
@@ -125,6 +126,9 @@ type searchMessageResponse struct {
 func (c *Client) Search(request SearchRequest) ([]*Message, error) {
 	var buf strings.Builder
 
+	if request.Query == "" && request.Filters == nil && request.MessageFilters == nil {
```

Comment:
You probably want a `len(request.Filters) == 0 && len(request.MessageFilters) == 0` here. Len works with the nil value—no need for an extra check.

Reply:
We can lint this.